v2/bundle/jwtbundle/bundle.go

```diff
@@ -1,68 +1,67 @@
 package jwtbundle
 
 import (
-	"bytes"
 	"crypto"
 	"encoding/json"
-	"errors"
-	"fmt"
 	"io"
 	"io/ioutil"
 	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/zeebo/errs"
 	"gopkg.in/square/go-jose.v2"
 )
 
+var (
+	jwtbundleErr = errs.Class("jwtbundle")
+)
+
 // Bundle is a collection of trusted JWT public keys for a trust domain.
 type Bundle struct {
-	mtx            sync.RWMutex
-	trustDomain    spiffeid.TrustDomain
-	jwtSigningKeys map[string]crypto.PublicKey
+	mtx         sync.RWMutex
+	trustDomain spiffeid.TrustDomain
+	jwtKeys     map[string]crypto.PublicKey
 }
 
 // New creates a new bundle.
 func New(trustDomain spiffeid.TrustDomain) *Bundle {
 	return &Bundle{
-		mtx:            sync.RWMutex{},
-		trustDomain:    trustDomain,
-		jwtSigningKeys: make(map[string]crypto.PublicKey),
+		trustDomain: trustDomain,
+		jwtKeys:     make(map[string]crypto.PublicKey),
 	}
 }
 
 // Load loads a Bundle from a file on disk.
 func Load(trustDomain spiffeid.TrustDomain, path string) (*Bundle, error) {
 	bundleBytes, err := ioutil.ReadFile(path)
-
 	if err != nil {
-		return nil, fmt.Errorf("unble to read JWT bundle: %w", err)
+		return nil, jwtbundleErr.New("unable to read JWT bundle: %w", err)
 	}
 
 	return Parse(trustDomain, bundleBytes)
 }
 
 // Read decodes a bundle from a reader.
 func Read(trustDomain spiffeid.TrustDomain, r io.Reader) (*Bundle, error) {
-	var b bytes.Buffer
-	_, err := b.ReadFrom(r)
+	b, err := ioutil.ReadAll(r)
 	if err != nil {
-		return nil, fmt.Errorf("unable to read: %v", err)
+		return nil, jwtbundleErr.New("unable to read: %v", err)
 	}
 
-	return Parse(trustDomain, b.Bytes())
+	return Parse(trustDomain, b)
 }
 
 // Parse parses a bundle from bytes.
 func Parse(trustDomain spiffeid.TrustDomain, bundleBytes []byte) (*Bundle, error) {
 	jwks := new(jose.JSONWebKeySet)
 	if err := json.Unmarshal(bundleBytes, jwks); err != nil {
-		return nil, fmt.Errorf("unable to parse JWK Set: %v", err)
+		return nil, jwtbundleErr.New("unable to parse JWKS: %v", err)
 	}
 
 	bundle := New(trustDomain)
 	for i, key := range jwks.Keys {
 		if err := bundle.AddJWTKey(key.KeyID, key.Key); err != nil {
-			return nil, fmt.Errorf("error adding entry %d of JWK Set: %v", i, err)
+			return nil, jwtbundleErr.New("error adding key %d of JWKS: %v", i, err)
```

Comment:
This ends up adding `jwtbundle` twice to the error message (see comment in the tests). We should unwrap the error (we lose the stack trace info for AddJWTKey but the error message is clear enough.

```suggestion
			return nil, jwtbundleErr.New("error adding key %d of JWKS: %v", i, errs.Unwrap(err))
```